pkg/storage/fs/posix/blobstore/blobstore.go

```diff
@@ -109,14 +116,17 @@ func (bs *Blobstore) Delete(node *node.Node) error {
 	return nil
 }
 
-func (bs *Blobstore) path(node *node.Node) (string, error) {
-	if node.BlobID == "" {
-		return "", fmt.Errorf("blobstore: BlobID is empty")
+func (bs *Blobstore) path(spaceID, blobID string) (string, error) {
```

Comment:
IMO the blobstore interface is way to constrained. We already have a hack in the ocis filestore implementation that 'moves' the file. s3ng could use a reader. posix might also want to move .. well it has to copy the bytes because of the user id.

Taking downloads into account we may want to forward range request headers to the actual s3 backend.

So maybe a parameter struct that carries the different properties makes sense?